.github/workflows/continuous-integration.yml

```diff
@@ -2,67 +2,111 @@
+  # discord
   DISCORD_URL_BUG: ${{ secrets.DISCORD_URL_BUG }}
   DISCORD_URL_FEEDBACK: ${{ secrets.DISCORD_URL_FEEDBACK }}
-
 jobs:
-  build:
+  pylint:
     runs-on: ubuntu-latest
-    strategy:
-      max-parallel: 4
-      matrix:
-        python-version: [3.8]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ env.PYTHON_TARGET }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_TARGET }}
+
+      - name: Install Python packages, excluding the unnecessary ones
+        run: |
+          python -m pip install --upgrade pip
+          sed -i '/\(tqdm\|gunicorn\|django\-heroku\|uwsgi\)/d' requirements.txt
+          pip install -r requirements.txt
+
+      - name: Run pylint
+        run: pylint --django-settings-module=${{env.DJANGO_SETTINGS_MODULE}} --load-plugins pylint_django tcf_website tcf_core
 
+  django:
+    runs-on: ubuntu-latest
+    outputs:
+      code-coverage: ${{ steps.coverage.outputs.percentage }}
     services:
       postgres:
         image: postgres:12.2
         env:
-          POSTGRES_USER: tcf_django
-          POSTGRES_PASSWORD: s3kr1t
-          POSTGRES_DB: tcf_db
+          POSTGRES_USER: ${{ env.DB_USER}}
+          POSTGRES_PASSWORD: ${{ env.DB_PASSWORD }}
+          POSTGRES_DB: ${{ env.DB_NAME }}
         ports:
           - 5432:5432
         # needed because the postgres container does not provide a healthcheck
         options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
-
     steps:
       - uses: actions/checkout@v2
 
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Set up Python ${{ env.PYTHON_TARGET }}
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: ${{ env.PYTHON_TARGET }}
 
-      - name: Install Python packages
+      - name: Install Python packages, excluding the unnecessary ones
         run: |
           python -m pip install --upgrade pip
+          sed -i '/\(pandas\|tqdm\|lint\|autopep8\|gunicorn\|django\-heroku\|uwsgi\)/d' requirements.txt
+          echo coverage >> requirements.txt
           pip install -r requirements.txt
 
-      - name: Run pylint
-        run: pylint --load-plugins pylint_django --django-settings-module=tcf_core.settings tcf_website tcf_core
-
       - name: Migrations & Tests
         run: |
           mkdir tcf_website/migrations
           touch tcf_website/migrations/__init__.py
+          envsubst < .config/.env.example > .env
           python manage.py makemigrations
           python manage.py migrate
-          python manage.py test
+          coverage run manage.py test
+        continue-on-error: true
+
+      - name: Get code coverage
+        if: ${{ success() }}
+        id: coverage
+        run: |
+          echo "::set-output name=percentage::$(coverage report | grep -o '[0-9]\+%' | tail -1)"
```

Comment:
`{0}` seems like an instance of [brace expansion](https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html). Just for your information, `grep -o '[0-9]+\%'` finds 1 or more decimal digits followed by a percent sign, and `tail -1` returns only the last occurrence.

Reply:
Right, but isn't the point of `echo` to print out the number? It doesn't appear to be part of the workflow log. In any case, the number seems to be getting passed to the Discord output so there's definitely something there (71% coverage seems a bit high though considering the low number of tests we have).

Reply:
I'm not sure if we should be looking for the output of `echo` in the log. The command is there because [that's how output is set, according to the GitHub Actions documentation](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter).